src/main/java/uk/ac/cam/cl/dtg/segue/api/UsersFacade.java

```diff
@@ -310,6 +310,73 @@ public Response getUserPreferences(@Context final HttpServletRequest httpServlet
         }
     }
 
+    /**
+     * An endpoint for group managers (often teachers) to send a password reset request email to group members without
+     * having to know the group members account email.
+     *
+     * @param request - request information used for caching
+     * @param httpServletRequest - the request, to work ou the current user
+     * @param userIdOfInterest - userId of interest - usually a the teacher's student
+     * @return a successful response regardless of whether the email exists or an error code if there is a technical
+     *         fault
+     */
+    @POST
+    @Path("users/{user_id}/resetpassword")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @GZIP
+    public Response generatePasswordResetTokenForOtherUser(@Context final Request request,
+                                                           @Context final HttpServletRequest httpServletRequest,
+                                                           @PathParam("user_id") final Long userIdOfInterest) {
+        try {
+            RegisteredUserDTO currentUser = userManager.getCurrentRegisteredUser(httpServletRequest);
+
+            RegisteredUserDTO userOfInterest = userManager.getUserDTOById(userIdOfInterest);
+            if (userOfInterest == null) {
+                throw new NoUserException("No user found with this ID.");
+            }
+
+            UserSummaryDTO userOfInterestSummaryObject = userManager.convertToUserSummaryObject(userOfInterest);
+
+            // decide if the user is allowed to view this data.
```

Comment:
Can we use this endpoint for admins too or do we already have one?

Reply:
This endpoint will work for admins because `userAssociationManager.hasPermission(...)` returns `True` for admins. 

However, there is no front-end way for admins to use it yet, partially because it is less useful for admins to create the token. The token is only valid for 24 hours and if the user doesn't read their emails until after the 24 hours it would likely cause more confusion than it's worth so it is usually easier to just ask them to request the reset email themselves when they are ready.

Reply:
This endpoint will work for admins because `userAssociationManager.hasPermission(...)` returns `True` for admins. 

However, there is no front-end way for admins to use it yet, partially because it is less useful for admins to create the token. The token is only valid for 24 hours and if the user doesn't read their emails until after the 24 hours it would likely cause more confusion than it's worth so it is usually easier to just ask them to request the reset email themselves when they are ready.